### PR TITLE
Towards universal QR codes

### PR DESCRIPTION
**Pull request by kristapsk**
Big step towards #53. Not a fan of such big changes in a single PR, but these were all kinda related.

Brief summary:

* Move Bitcoin invoice / QR code generation code out of individual nodes + add tests;
* Base class for all node classes (wasn't quite requirement, but seemed nice, now you have clear abstract class that shows minimum required to be implemented for new "node" backends);
* Split off `bolt11_invoice` from onchain `address` in `payments` table (db) [most important thing, this was what I just wanted to do when started doing this];
* More type hints.

Tested on my development laptop with onchain signet, as well as with [donate.kristapsk.lv](https://donate.kristapsk.lv/) (I will be reckless, this code is already running there). But more testing would be great, as there are a lots of LOCs changed.

Basically all the backend functionality needed for unified QR codes are IMHO there, rest remaining is changes in UI/UX and `satsale.py` part. Tried to only change backend things and keep everything the same from UX perspective for now.

For frontend part I actually think some switchable theme thing needs to be implemented, different people / use cases may want different frontends.

P.S. Why don't we have normal primary keys in db tables, etc, but maybe not so important and with current migration scheme it will be PITA.

### CODE REVIEW

**Comment by kristapsk** [comment]
This will conflict with #114, but will rebase one or another, depending which one gets merged first.

**Comment by kristapsk** [comment]
Updated `node/lndhub.py`.
